src/main/java/org/opentripplanner/graph_builder/module/ApplyMinTimeTransfers.java

```diff
@@ -0,0 +1,86 @@
+package org.opentripplanner.graph_builder.module;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import org.opentripplanner.graph_builder.DataImportIssueStore;
+import org.opentripplanner.graph_builder.services.GraphBuilderModule;
+import org.opentripplanner.model.PathTransfer;
+import org.opentripplanner.model.transfer.MinTimeTransfer;
+import org.opentripplanner.routing.graph.Graph;
+
+/**
+ * Transit data (for example transfers.txt) can describe how long a transfer between two stops can
+ * take. The previous graph build step {@link DirectTransferGenerator} calculates the transfers on
+ * the OSM street network and calculates a time, too.
+ * <p>
+ * In this step we take both data sources and combine them: if we already have an OSM-based transfer
+ * we use its edges in order to display a nice shape rather than a straight line, but we override
+ * with the one from the transit data.
+ */
+public class ApplyMinTimeTransfers implements GraphBuilderModule {
+
+    // https://en.wikipedia.org/wiki/Preferred_walking_speed
+    public final float AVERAGE_WALKING_SPEED = 1.4f;
+
+    @Override
+    public void buildGraph(
+            Graph graph,
+            HashMap<Class<?>, Object> extra,
+            DataImportIssueStore issueStore
+    ) {
+        Collection<MinTimeTransfer> minTimeTransfers =
+                (Collection<MinTimeTransfer>) extra.getOrDefault(
+                        MinTimeTransfer.class,
+                        Collections.EMPTY_LIST
+                );
+
+        minTimeTransfers.forEach(minTimeTransfer -> {
+            var transfers = graph.transfersByStop.get(minTimeTransfer.from);
+            var pathTransfers = transfers.stream()
+                    .filter(pathTransfer -> pathTransfer.to.getId()
+                            .equals(minTimeTransfer.to.getId()));
+
+            // we can have potentially multiple transfers if we compute them for bicycle or wheelchair users, for example,
+            // so we use the shortest and assume it's the walking one
+            var shortestTransfer =
+                    pathTransfers.min(Comparator.comparingDouble(PathTransfer::getDistanceMeters));
```

Comment:
The is a TODO on the PathTransfer:
```
 * TODO these should really have a set of valid modes in case bike vs. walk transfers are different
```
Would it be a big fix to add the modes so we you can filter by mode here instead?

Reply:
Not a huge change, so I just did it.

Since @flaktack wrote the code about multiple transfer modes, perhaps he could give a review.